### ISSUE TITLE
[13.0] active_test broken when added on field and view level

### ISSUE THREAD

**Issue by jeroen7s** [odoo/odoo#41939]
Impacted versions:
13.0

Steps to reproduce:
add One2many field with `context={'active_test': False}` to an Odoo model (sale.order in my case).
Add deactivated object to field (also sale.order in my case)

Current behavior:
no linked objects found if using `record.field_name` in python or in view
object is found if using `record.with_context(active_test=False).field_name`
object is also not found in views, despite adding `context="{'active_test': False}"` to the one2many field, the tree attribute and all underlying fields

Expected behavior:
active_test False should work

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
I don't get the problem and what do you mean with `record.field_name`.

**Comment by jeroen7s** [comment]
when you add a field like
```
 field_name = fields.One2many(
        comodel_name="sale.order",
        inverse_name="inverse",
        string="One2Many field",
        context={"active_test": False},
    )
```
you're supposed to be able to add deactivated objects (sale orders in this case) and still be able to see then in views / when using that field in code. This is not working.
if i run `record.field_name` I'm not seeing deactivated items that are added to that One2many field
and in the views i'm also unable to see deactivated sale orders on that field.

**Comment by pedrobaeza** [comment]
You are assuming something that is not true. The `one2many` only matches by inverse_name and domain, not by context. Do the following:

```python
    field_name = fields.One2many(
        comodel_name="sale.order",
        inverse_name="inverse",
        string="One2Many field",
        domain="['|', ('active', '=', False), ('active', '=', True)]",
    )
```

Closing as no valid issue, and with workaround provided.

**Comment by jeroen7s** [comment]
Thanks for the response, I guess I just made a wrong assumption. 
I do see it used in other places in Odoo on other One2many fields though
for example in the project module:
```
child_ids = fields.One2many('project.task', 'parent_id', string="Sub-tasks", context={'active_test': False})
```

**Comment by jeroen7s** [comment]
@pedrobaeza Adding the domain does not solve the problem
Sale orders added to the field still do not show up correctly if they are deactivated
![image](https://user-images.githubusercontent.com/10954827/70923032-79f2f380-2027-11ea-82c0-1f56b2b008f2.png)
![image](https://user-images.githubusercontent.com/10954827/70923056-870fe280-2027-11ea-9bfe-af5bcd4f52ef.png)
![image](https://user-images.githubusercontent.com/10954827/70923155-b3c3fa00-2027-11ea-8e8b-12d734f2f97b.png)

**Comment by jeroen7s** [comment]
@pedrobaeza both the domain and the context are set on the view, and both the context and the domain are set ont the field, yet still nothing is showing, while debugging the code, it clearly shows there are deactivated records linked to the field. (only visible when specifically running `record.with_context(active_test=False).field_name`)

**Comment by pedrobaeza** [comment]
Let's keep it opened then. I remember this to work in previous versions. I will give a try when possible.

**Comment by jeroen7s** [comment]
Did check again on an empty database, with only the base, sale, crm, installed and the module I'm porting from OCA where this issue appears. Should be easily reproducible. Small module with no dependencies:
https://github.com/OCA/sale-workflow/pull/1021
or if you prefer a zip of the module to reproduce it:
https://intuxikated.stackstorage.com/s/MrDjXgT1s5Io27T

**Comment by pedrobaeza** [comment]
This should be fixed in #42784

**Comment by rco-odoo** [comment]
Fixed by #42784
